### ISSUE TITLE
win: undeclared MiniDumpWithAvxXStateContext error in our mingw buildbots

### ISSUE THREAD

**Issue by santigimeno** [libuv/libuv#3992]
Since merging https://github.com/libuv/libuv/pull/3840 we're getting the following error in our mingw buildbots

```
/usr/bin/x86_64-w64-mingw32-gcc -DBUILDING_UV_SHARED=1 -DWIN32_LEAN_AND_MEAN -D_WIN32_WINNT=0x0602 -Duv_EXPORTS -I/home/runner/work/libuv/libuv/include -I/home/runner/work/libuv/libuv/src -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv.dir/src/win/process.c.obj -MF CMakeFiles/uv.dir/src/win/process.c.obj.d -o CMakeFiles/uv.dir/src/win/process.c.obj -c /home/runner/work/libuv/libuv/src/win/process.c
/home/runner/work/libuv/libuv/src/win/process.c: In function ‘uv__kill’:
/home/runner/work/libuv/libuv/src/win/process.c:1286:24: error: ‘MiniDumpWithAvxXStateContext’ undeclared (first use in this function)
 1286 |                        MiniDumpWithAvxXStateContext;
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/libuv/libuv/src/win/process.c:1286:24: note: each undeclared identifier is reported only once for each function it appears in
```

**Comment by bnoordhuis** [comment]
cc @staticfloat @vtjnash

**Comment by santigimeno** [comment]
`MinGW` support is Tier 3 so this issue should not be blocking the `v1.45.0` release but wanted to confirm we're ok with releasing without `MinGW` support. @libuv/collaborators

**Comment by bnoordhuis** [comment]
I don't think the functionality from #3840 is that valuable so I lean towards reverting if a fix doesn't show up quickly enough.

**Comment by gengjiawen** [comment]
It's actually fixed in mingw 12.2. Not sure we should patch it. (mingw 11.2 is not working from what I tested.)
